### PR TITLE
Add support for SUNDIALS 5.5-5.7 compiled with LAPACK

### PR DESCRIPTION
**Pull request by bryanwweber**
**Changes proposed in this pull request**

- Support for system SUNDIALS 5.5-5.7 with LAPACK. The preprocessor define indicating LAPACK support in SUNDIALS was changed with version 5.5.
- Fix that the Boost version message is printed even if Boost isn't found
- Simplify preprocessor logic in `BandMatrix.cpp`
- Fix deprecation warning about SunLinSol_SPGMR
- Fix error when `omp.h` is installed, but the library cannot be found by the linker (for example, omp installed by Homebrew, the Homebrew path on the search path, but using Apple's compilers).

**Checklist**

- [x] There is a clear use-case for this code change
- [x] The commit message has a short title & references relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] The pull request is ready for review

### CODE REVIEW

**Comment by bryanwweber** [comment]
By the way, I think it'd be good to backport at least ~64c0736~ fc93ece to the 2.5 branch

File: src/numerics/BandMatrix.cpp [hunk 1]
```diff
@@ -10,19 +10,11 @@
 #if CT_USE_LAPACK
     #include "cantera/numerics/ctlapack.h"
 #else
-    #if CT_SUNDIALS_USE_LAPACK
-        #if CT_SUNDIALS_VERSION >= 30
-            #include "sunlinsol/sunlinsol_lapackband.h"
-        #else
-            #include "cvodes/cvodes_lapack.h"
-        #endif
+    #if CT_SUNDIALS_VERSION >= 30
```
Comment (ischoegl) [inline src/numerics/BandMatrix.cpp:13]: the logic appears to be different here. The PR comment states that it was simplified ...

**Review by ischoegl — DISMISSED** [review]
Per my comments elsewhere, this isn't necessarily my strong suit (and I don't have the toolchains to verify this installed). On the other hand, both PR rationale and code comments are clear and the changes appear to be straightforward. Testing every combination in GH Actions may be overkill, but if there are candidates of particular concern this may be warranted. Overall, no concerns from my side ...

File: src/numerics/BandMatrix.cpp (hunk 1 above)
Comment (bryanwweber) [reply src/numerics/BandMatrix.cpp:13]: Hmmm good point. It's simplified in the sense that if `CT_USE_LAPACK` is false, then `CT_SUNDIALS_USE_LAPACK` is also always false, which means the first branch in the `# else` clause could never be followed. I'll clarify the commit message.

Comment (ischoegl) [reply src/numerics/BandMatrix.cpp:13]: Beyond, please feel free to ignore my 'Accept' - per discussion elsewhere, this is probably more a comment. *Edit: the commit message is likely adequate ...*

File: SConstruct [hunk 2]
```diff
@@ -1121,10 +1122,22 @@ if env['system_sundials'] == 'y':
...
+        # SUNLINSOL packages indicating whether SUNDIALS has been built with LAPACK
+        env["has_sundials_lapack"] = conf.CheckDeclaration(
+            "SUNDIALS_SUNLINSOL_LAPACKBAND",
+            '# include "sundials/sundials_config.h"',
+            "C++",
+        ) and conf.CheckDeclaration(
+            "SUNDIALS_SUNLINSOL_LAPACKDENSE",
+            '# include "sundials/sundials_config.h"',
+            "C++",
+        )
```
Comment (speth) [inline SConstruct]: Two formatting quibbles here:
* extraneous space between the `#` and `include` in each check
* I find this long `z = x and y` statement pretty hard to parse. I think it would be better to just break it up.

**Review by speth — COMMENTED** [review]
This looks good from a functional standpoint. I just have a couple of requests on the formatting.

**Review by speth — APPROVED** [review]
